src/components/HarvestQueueList.js

```diff
@@ -13,22 +14,22 @@ import {
   NuGetVersionPicker,
   RubyGemsVersionPicker
 } from './'
+import { getGitHubRevisions } from '../api/clearlyDefined'
```

Comment:
this should take a function like `GitHubCommitPicker` so we don't have to grab the token from the store. Would be cleaner and more consistent.

Reply:
Accessing the store has to go somewhere and the further up it goes, the more we have to pass props through layers. Where do you want to put it? For HarvestQueueList the very next level up is a connected component, but for SourcePicker you have three layers on top of both call places:

```
AbstractPageDefinitions
ComponentList
DefinitionEntry
```

and

```
AbstractFullDetailsView
FullDetailComponent
DescribedSection
```

I'll be happy to refactor to whatever, but I don't see which configuration would be better than the current.

Reply:
Accessing the store has to go somewhere and the further up it goes, the more we have to pass props through layers. Where do you want to put it? For HarvestQueueList the very next level up is a connected component, but for SourcePicker you have three layers on top of both call places:

```
AbstractPageDefinitions
ComponentList
DefinitionEntry
```

and

```
AbstractFullDetailsView
FullDetailComponent
DescribedSection
```

I'll be happy to refactor to whatever, but I don't see which configuration would be better than the current.